dashboard/app/controllers/home_controller.rb

```diff
@@ -86,31 +86,24 @@ def init_homepage
       @sections = current_user.sections.map(&:summarize)
       @student_sections = current_user.sections_as_student.map(&:summarize)
 
-      # Students will receive a @student_top_course for their primary script,
-      # so we don't want to include that script (if it exists) in the regular
-      # lists of recent scripts.
-      exclude_primary_script = current_user.student?
+      # Students and teachers will receive a @top_course for their primary script, so we don't want to include that script (if it exists) in the regular lists of recent scripts.
+      exclude_primary_script = current_user
```

Comment:
I also wonder if this API of passing a top course and other courses still makes sense in a world where both student and teacher use top course. I wonder if we can simplify things. Maybe pass a list of all courses, and an index of which one is the "top" course, and then let the client determine what to do with these.